### ISSUE TITLE
[Disney+]

### ISSUE THREAD

**Issue by Skaalgard** [matthuisman/slyguy.addons#236]
I'm running LibreELEC on a Raspberry Pi 4. I've installed the latest version of Disney+ from the SlyGuy repository, and when I play anything, I just have audio, no video at all, and if I let it play long enough my whole system crashes. I've seen various solutions elsewhere to this and none of them have worked. Please help.

**Comment by matthuisman** [comment]
Can you please provide a Kodi debug log

**Comment by Ghosties-Kodi** [comment]
I have same problem. Audio works. Video only pixels, no picture. Testet with new Marvel Dr.Strange

**Comment by matthuisman** [comment]
Still need a full kodi debug log please 
https://kodi.wiki/view/Log_file/Easy

**Comment by scottydulton** [comment]
Try un ticking the 4K box in settings

**Comment by josh-homerston** [comment]
I've been having the same issues on OSMC 2022.03-1 (Linux K: 5.10.78-5-osmc) running on top of a Raspberry Pi 3 Model B+. The OSMC interface fails to expose a 4K enable checkbox from what I can see...

I will attempt to generate a Kodi log file tomorrow, when I have some spare time.

**Comment by matthuisman** [comment]
@josh-homerston your problem is
```
CDM LoadNativeLibrary error: /home/osmc/.kodi/cdm/libwidevinecdm.so: cannot open shared object file: No such file or directory
Unable to load widevine shared library (/home/osmc/.kodi/cdm/libwidevinecdm.so)
```
Go to Disney+ addon > Settings > Advanced > (Re)install Widevine CDM

**Comment by josh-homerston** [comment]
> @josh-homerston your problem is
> 
> ```
> CDM LoadNativeLibrary error: /home/osmc/.kodi/cdm/libwidevinecdm.so: cannot open shared object file: No such file or directory
> Unable to load widevine shared library (/home/osmc/.kodi/cdm/libwidevinecdm.so)
> ```
> 
> Go to Disney+ addon > Settings > Advanced > (Re)install Widevine CDM

No joy sadly... Tried both Widevine V4.10.2449.0 and V4.10.2252.5, and after each installation I reset the plugin from the menu, then power-cycled the Pi from the Power options in the menu, and re-logged in.

Still, typically just the audio plays... Sometimes when selecting a higher quality stream, the play progress counter increments with neither video or audio.

Might I have more luck on LibreELEC, is this likely to be an OSMC issue?

**Comment by matthuisman** [comment]
Hmmm, I wonder if there is a newer required widevine lib for arm

**Comment by matthuisman** [comment]
If you go back into install widevine, does it detect one is installed?

**Comment by josh-homerston** [comment]
Yeah, it seems to, and it changes correctly if I install a different version. The plugin certainly seems to think the files are installed correctly, installation process exits with the message 'Widevine CDM (x.xx.xxxx.x) Installed successfully'.

**Comment by matthuisman** [comment]
And if you view a new log, is that same file not found error there?

**Comment by josh-homerston** [comment]
The logfile doesn't seem to contain the error any more, but the same issue persists. New log; https://bluishsmoke.uk/temp/kodi.log2.txt

**Comment by matthuisman** [comment]
Unable to load widevine shared library (/home/osmc/.kodi/cdm/libwidevinecdm.so)

Interesting. 
You using any other add-ons that use widevine? Eg. Netflix

**Comment by josh-homerston** [comment]
Not to my knowledge, pretty lean install tbh. Got the Spotify add-on installed, other than that, it's just a media file browser for video and music on my NAS.

**Comment by matthuisman** [comment]
i have checked and 4.10.2449.0 is indeed the latest widevine version.
So unsure sorry. Maybe a permission issue. 
Maybe OSMC forgot to keep the libc6 patch in that was required to work with newer widevine libs?
@samnazarko any ideas?

**Comment by samnazarko** [comment]
The libc6 patch is still in. It would be good to get some logs from the user.

First thing I would do is check that the libwidevine so is the correct arch (ie armv7). We haven’t had reports of issues on the forums..

**Comment by josh-homerston** [comment]
When downloading a Widevine CDM version through the plugin's settings menu, the file name is reported as 'x.xx.xxxx.x-linux-armv7.so'.

**Comment by matthuisman** [comment]
@samnazarko log here
https://bluishsmoke.uk/temp/kodi.log.txt
I wonder if it's some sort of sandpit where a .so can't be used? It does say Linux (thumb) which o havn't seen before

**Comment by samnazarko** [comment]
I’d suggest the user tries the Bullseye version

On 30 Jul 2022, at 18:58, Matt Huisman ***@***.***> wrote:

﻿

@samnazarko<https://github.com/samnazarko> log here
https://bluishsmoke.uk/temp/kodi.log.txt
I wonder if it's some sort of sandpit where a .so can't be used? It does say Linux (thumb) which o havn't seen before

**Comment by josh-homerston** [comment]
> I’d suggest the user tries the Bullseye version

Just spent a few run-up's at installing this; And for plenty of attempts can't get the WiFi to connect. That's using the installer, both providing credentials through that interface, and also attempted selecting 'wired' in the wizard, and then after install attempting to configure the WiFi. Fails to connect immediately, no delay at all after providing a password.

**Comment by matthuisman** [comment]
Please post osmc issues in the osmc forum. They may get missed here

**Comment by josh-homerston** [comment]
> Please post osmc issues in the osmc forum. They may get missed here

Apologies, my bad, lack of coffee to blame!
